studio/components/SalaryCsvInput/SalariesInput.tsx

```diff
@@ -0,0 +1,144 @@
+import { SalariesParseErrorsToastDescription } from "./SalariesParseErrorsToastDescription";
+import SalaryNumberInput from "./SalaryNumberInput";
+
+const UPLOAD_CSV_INPUT_ID = "upload-csv-input";
+
+export const SalariesInput = (props: StringInputProps) => {
+  const toast = useToast();
+
+  const salaries =
+    props.value === undefined
+      ? undefined
+      : salariesFromStoredString(props.value);
+
+  async function handleFileRead(e: ProgressEvent<FileReader>): Promise<void> {
+    const fileData = e.target?.result;
+    if (fileData === null || typeof fileData !== "string") {
+      toast.push({
+        title: "Invalid data",
+        description: "Verify the file content and try again.",
+        status: "error",
+      });
+      return;
+    }
+    const salariesParseResult = salariesFromCsvString(fileData);
+    if (!salariesParseResult.ok) {
+      toast.push({
+        title: "Invalid salaries data",
+        description: (
+          <SalariesParseErrorsToastDescription
+            errors={salariesParseResult.error}
+          />
+        ),
+        status: "error",
+      });
+      return;
+    }
+    props.onChange(set(salariesAsStoredString(salariesParseResult.value)));
+  }
+
+  function handleFileChange(e: ChangeEvent<HTMLInputElement>): void {
+    if (e.target.files !== null && e.target.files.length > 0) {
+      const file = e.target.files[0];
+      if (file.type === "text/csv") {
+        const reader = new FileReader();
+        reader.onload = handleFileRead;
+        reader.readAsText(file);
+      } else {
+        toast.push({
+          title: "Invalid salaries file type",
+          description: "Verify the file extension and try again.",
+          status: "error",
+        });
+      }
+    }
+    // reset to allow subsequent uploads of the same file
+    e.target.value = "";
+  }
+
+  function handleYearSalaryChange(year: string, salary: number): void {
+    props.onChange(
+      set(
+        salariesAsStoredString({
+          ...salaries,
+          [year]: salary,
+        }),
+      ),
+    );
+  }
+
+  return (
+    <Stack space={4}>
+      <Inline space={2}>
+        <div>
+          {/*
+            Using label for hidden input as a custom file upload button, based on:
+            https://developer.mozilla.org/en-US/docs/Web/API/File_API/Using_files_from_web_applications#using_a_label_element_to_trigger_a_hidden_file_input_element
+          */}
+          <input
+            id={UPLOAD_CSV_INPUT_ID}
+            className={styles.uploadInput}
+            type={"file"}
+            onChange={handleFileChange}
+            accept=".csv"
+          />
+          <Box className={styles.uploadButtonWrapper}>
+            <button className={styles.uploadButton} tabIndex={-1}>
+              <label
+                htmlFor={UPLOAD_CSV_INPUT_ID}
+                className={styles.uploadButtonContent}
+              >
+                <UploadIcon className={styles.uploadButtonIcon} />
```

Comment:
```suggestion
                <UploadIcon className={styles.uploadButtonIcon} />. //use as an :after in label instead, if possible, so to avoid any access to the icon my SRs
```

Reply:
Is it not ignored already, since it has no alt text?

Reply:
By most yes, but some SRs do read them anyway. It's also nice practice to add icons such as these as css pseudo elements when they're meant for decoration only.

Reply:
Can you give an example of this? Would it mean we have to include the individual SVGs in the source code, instead of importing directly from the Sanity package?

Reply:
no can doo.. didn't think about the fact that css modules cannot have imported svgs. 🙃